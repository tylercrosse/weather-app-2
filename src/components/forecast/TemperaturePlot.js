/* eslint-disable array-callback-return */

import React from 'react';
import PropTypes from 'prop-types'
import {
  XYPlot,
  YAxis,
  HorizontalGridLines,
  VerticalGridLines,
  AreaSeries,
  LineSeries,
  Crosshair
} from 'react-vis';

/**
 * Format items for Crosshair.
 * @param {Array} values Array of values.
 * @returns {*} Formatted list of items.
 */
function itemsFormat(values) {
  return values.map((v, i) => {
    if (v) {
      return { value: v.y, title: '°F' };
    }
  });
}

const TemperaturePlot = props => {
  return (
    <XYPlot
      animation
      className="forecast__chart chart__temps"
      height={200}
      width={props.chartWidth}
      onMouseLeave={props.handleMouseLeave}
    >
      <YAxis hideLine left={6} tickFormat={v => `${v}°F`} />
      <HorizontalGridLines />
      <VerticalGridLines tickValues={props.dayDivsions} />
      <AreaSeries
        data={props.tempsNightData}
        style={{
          stroke: 'none',
          fill: 'rgba(130, 130, 130, 0.1)'
        }}
      />
      <LineSeries
        color="#FC0F45"
        curve="curveMonotoneX"
        data={props.tempsData}
        onNearestX={props.handleNearestX}
      />
      <LineSeries
        color="grey"
        opacity={0.4}
        data={[
          { x: props.currentTime, y: props.tempsRange.min },
          { x: props.currentTime, y: props.tempsRange.max }
        ]}
      />
      <Crosshair
        values={props.crosshairValues}
        titleFormat={props.titleFormat}
        itemsFormat={itemsFormat}
      />
    </XYPlot>
  );
};

TemperaturePlot.propTypes = {
  chartWidth: PropTypes.number.isRequired,
  crosshairValues: PropTypes.array.isRequired,
  handleMouseLeave: PropTypes.func.isRequired,
  handleNearestX: PropTypes.func.isRequired,
  dayDivsions: PropTypes.array.isRequired,
  titleFormat: PropTypes.func.isRequired,
  currentTime: PropTypes.number.isRequired,
  tempsData: PropTypes.array.isRequired,
  tempsNightData: PropTypes.array.isRequired,
  tempsRange: PropTypes.object.isRequired,
}

export default TemperaturePlot;
