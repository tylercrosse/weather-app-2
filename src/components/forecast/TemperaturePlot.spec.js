import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import TemperaturePlot from './TemperaturePlot';

const setup = () => {
  const props = {
    chartWidth: 880,
    crosshairValues: [],
    handleMouseLeave: jest.fn(),
    handleNearestX: jest.fn(),
    dayDivsions: [],
    titleFormat: jest.fn(),
    currentTime: 1499105000,
    tempsData: [],
    tempsNightData: [],
    tempsRange: {},
  }

  const component = <TemperaturePlot {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<TemperaturePlot />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
