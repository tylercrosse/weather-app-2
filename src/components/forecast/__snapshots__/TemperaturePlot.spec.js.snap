// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemperaturePlot /> should render correctly 1`] = `
<XYPlot
  animation={true}
  className="forecast__chart chart__temps"
  height={200}
  onMouseLeave={[Function]}
  width={880}
>
  <YAxis
    attr="y"
    hideLine={true}
    left={6}
    orientation="left"
    tickFormat={[Function]}
  />
  <HorizontalGridLines
    attr="y"
    direction="horizontal"
  />
  <VerticalGridLines
    attr="x"
    direction="vertical"
    tickValues={Array []}
  />
  <AreaSeries
    className=""
    data={Array []}
    style={
      Object {
        "fill": "rgba(130, 130, 130, 0.1)",
        "stroke": "none",
      }
    }
  />
  <LineSeries
    color="#FC0F45"
    curve="curveMonotoneX"
    data={Array []}
    onNearestX={[Function]}
    opacity={1}
    strokeStyle="solid"
    style={Object {}}
  />
  <LineSeries
    color="grey"
    curve={null}
    data={
      Array [
        Object {
          "x": 1499105000,
          "y": undefined,
        },
        Object {
          "x": 1499105000,
          "y": undefined,
        },
      ]
    }
    opacity={0.4}
    strokeStyle="solid"
    style={Object {}}
  />
  <Crosshair
    itemsFormat={[Function]}
    titleFormat={[Function]}
    values={Array []}
  />
</XYPlot>
`;
